cfgmgr/buffer_pool_mellanox.lua

```diff
@@ -165,7 +196,14 @@ for i = 1, #epools, 1 do
     end
 end
 
+if shp_enabled and shp_size == 0 then
+    accumulative_xoff = math.ceil(accumulative_xoff / over_subscribe_ratio)
+else
+    accumulative_xoff = shp_size
+end
+
 local pool_size
+accumulative_occupied_buffer = accumulative_occupied_buffer + accumulative_xoff
```

Comment:
3 cases:
- SHP enabled by over subscribe ratio, `accumulative_xoff` represents SHP size
- SHP enabled by size, `accumulative_xoff` represents SHP size
- SHP disabled, `accumulative_xoff` represents the accumulative xoff